nequip/nn/_graph_mixin.py

```diff
@@ -266,8 +285,28 @@ def insert(
             idx += 1
         names.insert(idx, name)
         modules.insert(idx, module)
+
         self._modules = OrderedDict(zip(names, modules))
-        # TODO: handle irreps
+
+        module_list = list(self._modules.values())
+
+        # sanity check the compatibility
+        if idx > 0:
+            assert AtomicDataDict._irreps_compatible(
+                module_list[idx - 1].irreps_out, module.irreps_in
+            )
+        if len(module_list) > idx:
+            assert AtomicDataDict._irreps_compatible(
+                module_list[idx + 1].irreps_in, module.irreps_out
+            )
+
+        # insert the new irreps_out to the later modules
+        for module_id, next_module in enumerate(module_list[idx + 1 :]):
+            next_module.add_independent_irreps(module.irreps_out)
+
+        # update the final wrapper irreps_out
+        self.irreps_out = dict(module_list[-1].irreps_out)
```

Comment:
why is `dict(` needed

Reply:
I don't know. I was copying from your append method :woman_facepalming: 

`def append(self, name: str, module: GraphModuleMixin) -> None:
          r"""Append a module to the SequentialGraphNetwork.
  
          Args:
              name (str): the name for the module
              module (GraphModuleMixin): the module to append
          """
>>        assert AtomicDataDict._irreps_compatible(self.irreps_out, module.irreps_in)
          self.add_module(name, module)
          self.irreps_out = dict(module.irreps_out)
          return
`

Reply:
nice, it can't break anyuthng so its fine